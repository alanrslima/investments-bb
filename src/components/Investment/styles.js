import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    backgroundColor: colors.WHITE,
    padding: 20,
    flexDirection: 'row',
    borderBottomWidth: 2,
    borderBottomColor: colors.GRAY,
  },
  containerInfo: {
    flex: 0.6,
  },
  containerValue: {
    flex: 0.4,
    alignItems: 'flex-end',
  },
  title: {
    color: colors.BLACK,
    fontWeight: '600',
    fontSize: fonts.TITLE,
  },
  description: {
    color: colors.GRAY_DARK,
    fontSize: fonts.DESCRIPTION,
    marginTop: 2,
  },
  value: {
    color: colors.BLACK,
    fontWeight: '600',
    fontSize: fonts.DESCRIPTION,
  }
});
