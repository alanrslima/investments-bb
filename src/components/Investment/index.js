import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

function Investment({
  title, description, value, onPress, disabled
}) {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      disabled={disabled}
    >
      <View style={styles.containerInfo}>
        <Text style={[styles.title, disabled && styles.disabled]}>{title}</Text>
        <Text style={[styles.description, disabled && styles.disabled]}>{description}</Text>
      </View>
      <View style={styles.containerValue}>
        <Text
          numberOfLines={1}
          style={[styles.value, disabled && styles.disabled]}
        >
          {value}
        </Text>
      </View>
    </TouchableOpacity>
  );
}

Investment.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

Investment.defaultProps = {
  disabled: false,
};

export default Investment;
