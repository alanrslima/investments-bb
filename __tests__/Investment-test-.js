import React from 'react';
import renderer from 'react-test-renderer';
import Investment from '../src/components/Investment';


test('renders correctly', () => {
  const tree = renderer.create(<Investment
    onPress={() => { }}
    description="Minha aposentadoria"
    title="Investimento I"
    value="R$ 26.000,00"
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
