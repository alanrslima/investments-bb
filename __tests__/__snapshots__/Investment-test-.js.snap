// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#F4F4F4",
      "borderBottomWidth": 2,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.6,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "600",
          },
          false,
        ]
      }
    >
      Investimento I
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#707070",
            "fontSize": 15,
            "marginTop": 2,
          },
          false,
        ]
      }
    >
      Minha aposentadoria
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.4,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 15,
            "fontWeight": "600",
          },
          false,
        ]
      }
    >
      R$ 26.000,00
    </Text>
  </View>
</View>
`;
